xla/hlo/transforms/simplifiers/algebraic_simplifier_test.cc

```diff
@@ -12688,5 +12688,31 @@ TEST_F(AlgebraicSimplifierTest, TestNew123) {
   EXPECT_FALSE(simplifier.Run(module.get()).value());
 }
 
+TEST_F(AlgebraicSimplifierTest,
+       ReducePrecisionWithSameDatatypeShouldBeRemoved) {
+  const char* hlo = R"(
+  HloModule test
+  ENTRY main {
+    param.0 = bf16[2,16,64]{2,1,0} parameter(0)
+    param.1 = bf16[32,64]{1,0} parameter(1)
+    reduce-precision.46 = bf16[32,64]{1,0} reduce-precision(param.1), exponent_bits=8, mantissa_bits=7
+    bitcast.7734 = bf16[2,16,64]{2,1,0} bitcast(reduce-precision.46)
```

Comment:
Do we need this many instructions to exercise the test? I would have assumed that

```
p0 = bf16[64] parameter(0)
ROOT reduce-precision = bf16[64] reduce-precision(p0), exponent_bits=8, mantissa_bits=7
```

Would have been sufficient?